lib/auth/test-functions.tsx

```diff
@@ -0,0 +1,92 @@
+import { sign } from 'jsonwebtoken';
+import { NextApiRequest } from 'next';
+import { ParsedUrlQuery } from 'querystring';
+import { ServerResponse } from 'http';
+
+export const dateToUnix = (date: Date): number =>
+  Math.floor(date.getTime() / 1000);
+
+interface MakeTokenInput {
+  sub?: string;
+  email?: string;
+  iss?: string;
+  name?: string;
+  groups?: Array<string>;
+  iat?: Date;
+}
+
+export const makeToken = ({
+  sub = '49516349857314',
+  email = 'test@example.com',
+  iss = 'Hackney',
+  name = 'example user',
+  groups = ['test-group'],
+  iat = new Date(),
+}: MakeTokenInput): string =>
+  sign(
+    { sub, email, iss, name, groups, iat: dateToUnix(iat) },
+    process.env.HACKNEY_AUTH_TOKEN_SECRET as string
+  );
+
+export interface MakeNextApiRequestInput {
+  method?:
+    | 'GET'
+    | 'HEAD'
+    | 'POST'
+    | 'PUT'
+    | 'DELETE'
+    | 'CONNECT'
+    | 'OPTIONS'
+    | 'TRACE'
+    | 'PATCH';
+  query?: ParsedUrlQuery;
+  url?: string;
+  body?: unknown;
+  headers?: {
+    [key: string]: string;
+  };
+  cookies?: {
+    [key: string]: string;
+  };
+}
+
+export const makeNextApiRequest = ({
```

Comment:
I think this is the only method needed from this file, might be a good refactor to move it into the spec file for csrf tokens and get rid of the rest of the file.